pkg/pgmodel/end_to_end_tests/promql_query_endpoint_test.go

```diff
@@ -272,59 +297,2693 @@ func TestPromQLQueryEndpoint(t *testing.T) {
 			name:  "complex query 3",
 			query: `holt_winters(metric_1[10m], 0.1, 0.5)`,
 		},
-	}
-
-	withDB(t, *testDatabase, func(db *pgxpool.Pool, t testing.TB) {
-		// Ingest test dataset.
-		ingestQueryTestDataset(db, t, generateLargeTimeseries())
-		// Getting a read-only connection to ensure read path is idempotent.
-		readOnly := testhelpers.GetReadOnlyConnection(t, *testDatabase)
-		defer readOnly.Close()
-
-		var tester *testing.T
-		var ok bool
-		if tester, ok = t.(*testing.T); !ok {
-			t.Fatalf("Cannot run test, not an instance of testing.T")
-			return
-		}
-
-		r := pgmodel.NewPgxReader(readOnly, nil, 100)
-		queryable := query.NewQueryable(r.GetQuerier())
-		queryEngine := query.NewEngine(log.GetLogger(), time.Minute)
-
-		apiConfig := &api.Config{}
-		instantQuery := api.Query(apiConfig, queryEngine, queryable)
-		rangeQuery := api.QueryRange(apiConfig, queryEngine, queryable)
-
-		apiURL := fmt.Sprintf("http://%s:%d/api/v1", testhelpers.PromHost, testhelpers.PromPort.Int())
-		client := &http.Client{Timeout: 10 * time.Second}
-
-		start := time.Unix(startTime/1000, 0)
-		end := time.Unix(endTime/1000, 0)
-		var (
-			requestCases []requestCase
-			req          *http.Request
-			err          error
-		)
-		for _, c := range testCases {
-			req, err = genInstantRequest(apiURL, c.query, start)
-			if err != nil {
-				t.Fatalf("unable to create PromQL query request: %s", err)
-			}
-			requestCases = append(requestCases, requestCase{req, fmt.Sprintf("%s (instant query, ts=start)", c.name)})
-
-			// Instant Query, 30 seconds after start
-			req, err = genInstantRequest(apiURL, c.query, start.Add(time.Second*30))
-			if err != nil {
-				t.Fatalf("unable to create PromQL query request: %s", err)
-			}
-			requestCases = append(requestCases, requestCase{req, fmt.Sprintf("%s (instant query, ts=start+30sec)", c.name)})
-		}
-		testMethod := testRequestConcurrent(requestCases, instantQuery, client, queryResultComparator)
-		tester.Run("test instant query endpoint", testMethod)
-
-		requestCases = nil
-		for _, c := range testCases {
+		{
```

Comment:
Let's acknowledge in comments where these queries came from and a link to the repo